app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,51 @@
+require_relative "../lib/oidc_client.rb"
+
+class SessionsController < ApplicationController
+  def create
+    if logged_in?
+      redirect_to redirect_path
+      return
+    end
+
+    redirect_to oidc.auth_uri[:uri]
+  end
+
+  def callback
+    unless params[:code]
+      redirect_to redirect_path
+      return
+    end
+
+    state = params.require(:state)
+
+    user_info = oidc.callback(
+      params.require(:code),
+      state,
+    )
+
+    session[:sub] = user_info.sub
+    session[:email] = user_info.email
+
+    redirect_to redirect_path
+  end
+
+  def delete
+    session.delete(:sub)
+    session.delete(:email)
+    redirect_to redirect_path
+  end
+
+private
+
+  def redirect_path
+    transition_checker_questions_path
+  end
+
+  def oidc
+    @oidc ||= OidcClient.new(
+      ENV["ACCOUNT_MANAGER_URL"],
```

Comment:
Could we get this out of Plek instead?
```suggestion
      Plek.find("account-manager"),
```

Reply:
I had changed this in a later commit (because the `accounts_api` service wasn't there when I started writing this code).  I've rebased so we use Plek from the start, which has removed one of the commits.